modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/dcrm/api/OAuthDCRMTestCase.java

```diff
@@ -117,6 +117,44 @@ public void testCreateServiceProviderRequest() throws IOException {
 
     }
 
+    @Test(alwaysRun = true, groups = "wso2.is", priority = 7, description =
+            "Create a service provider with saas app property")
+    public void testCreateServiceProviderRequestWithSaasApp() throws IOException {
+        HttpPost request = new HttpPost(getPath());
+        request.addHeader(HttpHeaders.AUTHORIZATION, getAuthzHeader());
+        request.addHeader(HttpHeaders.CONTENT_TYPE, OAuthDCRMConstants.CONTENT_TYPE);
+
+        JSONArray grantTypes = new JSONArray();
+        grantTypes.add(OAuthDCRMConstants.GRANT_TYPE_AUTHORIZATION_CODE);
+        grantTypes.add(OAuthDCRMConstants.GRANT_TYPE_IMPLICIT);
+
+        JSONArray redirectURI = new JSONArray();
+        redirectURI.add(OAuthDCRMConstants.REDIRECT_URI);
+
+        JSONObject obj = new JSONObject();
+        obj.put(OAuthDCRMConstants.CLIENT_NAME, OAuthDCRMConstants.APPLICATION_NAME_WITH_SAAS);
+        obj.put(OAuthDCRMConstants.GRANT_TYPES, grantTypes);
+        obj.put(OAuthDCRMConstants.REDIRECT_URIS, redirectURI);
+        obj.put(OAuthDCRMConstants.EXT_SAAS_APP, true);
+
+        StringEntity entity = new StringEntity(obj.toJSONString());
+
+        request.setEntity(entity);
+
+        HttpResponse response = client.execute(request);
+        assertEquals(response.getStatusLine().getStatusCode(), 201, "Service Provider " +
+                "has not been created successfully");
+
+        BufferedReader rd = new BufferedReader(new InputStreamReader(response.getEntity().getContent()));
+
+        Object responseObj = JSONValue.parse(rd);
+        EntityUtils.consume(response.getEntity());
+        client_id = ((JSONObject) responseObj).get("client_id").toString();
+
+        assertNotNull(client_id, "client_id cannot be null");
```

Comment:
Do we get the value of SaaS property sent in the DCR response?
If we do, then better to assert it.

Reply:
No, Since its optional value I do not add it to the response object.